src/Razor/Microsoft.NET.Sdk.Razor/src/CreateBlazorTrimmerRootDescriptorFile.cs

```diff
@@ -3,42 +3,53 @@
 
 using System.Collections.Generic;
 using System.IO;
-using System.Linq;
 using System.Xml;
 using System.Xml.Linq;
 using Microsoft.Build.Framework;
 using Microsoft.Build.Utilities;
 
-namespace Microsoft.AspNetCore.Components.WebAssembly.Build
+namespace Microsoft.AspNetCore.Razor.Tasks
 {
     // Based on https://github.com/mono/linker/blob/3b329b9481e300bcf4fb88a2eebf8cb5ef8b323b/src/ILLink.Tasks/CreateRootDescriptorFile.cs
-    public class BlazorCreateRootDescriptorFile : Task
+    public class CreateBlazorTrimmerRootDescriptorFile : Task
     {
         [Required]
-        public ITaskItem[] AssemblyNames { get; set; }
+        public ITaskItem[] Assemblies { get; set; }
 
         [Required]
-        public ITaskItem RootDescriptorFilePath { get; set; }
+        public ITaskItem TrimmerFile { get; set; }
 
         public override bool Execute()
         {
-            using var fileStream = File.Create(RootDescriptorFilePath.ItemSpec);
-            var assemblyNames = AssemblyNames.Select(a => a.ItemSpec);
+            using var fileStream = File.Create(TrimmerFile.ItemSpec);
 
-            WriteRootDescriptor(fileStream, assemblyNames);
+            WriteRootDescriptor(fileStream);
             return true;
         }
 
-        internal static void WriteRootDescriptor(Stream stream, IEnumerable<string> assemblyNames)
+        internal void WriteRootDescriptor(Stream stream)
         {
             var roots = new XElement("linker");
-            foreach (var assemblyName in assemblyNames)
+            foreach (var assembly in Assemblies)
             {
+                var assemblyName = assembly.GetMetadata("FileName") + assembly.GetMetadata("Extension");
```

Comment:
We should be able to remove these trimmer files once https://github.com/dotnet/sdk/issues/12035 lands.

Reply:
https://github.com/dotnet/sdk/issues/12035 has a lot of ongoing discussion about what is and isn't in scope, so it's hard for me to be sure what it is/isn't going to do for us in the end.

With the implementation you have here, it looks like you're retaining the type-granularity linking we did before. Is that correct? If so that's great - we may well end up needing to keep this logic.

Reply:
Yup, we have the type level granularity from before. They are aware of what we are doing, so my hope if we get something cheap.